src/codegate/db/connection.py

```diff
@@ -416,6 +416,16 @@ async def soft_delete_workspace(self, workspace: WorkspaceRow) -> Optional[Works
         return deleted_workspace
 
     async def hard_delete_workspace(self, workspace: WorkspaceRow) -> Optional[WorkspaceRow]:
+        # First delete associated muxes
+        sql_delete_muxes = text(
+            """
+            DELETE FROM muxes
+            WHERE workspace_id = :id
+            """
+        )
+        await self._execute_with_no_return(sql_delete_muxes, {"id": workspace.id})
```

Comment:
Is this needed? Muxes should have an `ON DELETE CASCADE` that would clean them up if a workspace is hard-deleted.